### ISSUE TITLE
Strange rotation of images

### ISSUE THREAD

**Issue by danvonk** [LaurentRDC/hakyll-images#11]
Hi,
firstly thanks for the library however I am experiencing some strangeness: I have a few pictures on my hakyll site that are taller than they are wide and running them through the `compressJpgCompiler` seems to rotate them? I am using version `1.2.0`

For example, below, the original image is on the left-hand side and the compressed one, found in `_site` is on the right:
![imagewrongway](https://github.com/LaurentRDC/hakyll-images/assets/966703/61da6156-48bd-470f-b52e-55626cb7283e)

My route in site.hs looks like:
```haskell
      forM_ ["images/**.JPG", "images/**.jpg"] $ \f -> match f $ do
        route idRoute
        compile $ loadImage >>= compressJpgCompiler 50
```
and the exmaple image used is:
![14460014](https://github.com/LaurentRDC/hakyll-images/assets/966703/a87afa3e-13bc-4323-aa4f-db0f4f2cff42)

Thanks in advance

**Comment by LaurentRDC** [comment]
Hmmm that is strange. 

All image manipulations (including compression) is handled by the library `JuicyPixels`. In particular, the function which does the compression is [`imageToJpg`](https://hackage.haskell.org/package/JuicyPixels-3.3.8/docs/Codec-Picture-Saving.html#v:imageToJpg). I would guess that this is where the problem is?

Let me play around and see what's going on

**Comment by LaurentRDC** [comment]
I believe that this is due to metadata of the original image being stripped. I should be able to create a new release today which fixes this issue.

**Comment by danvonk** [comment]
Ah great, that would be awesome! It's not urgent, so no worries on the timing :)

**Comment by LaurentRDC** [comment]
I believe I have fixed the problem, and everything is working as expected on my personal site.

Before I make a release, would you be able to test the new version on your site?

You can do so by adding the following clause in your `cabal.project` file:

```txt
source-repository-package
  type:     git
  location: https://github.com/LaurentRDC/hakyll-images
  tag:      d0833c399f9bb14a35fbe33d4452eda2eeaf2d95
```

**Comment by danvonk** [comment]
Sure. I tested using that commit and the images aren't being rotated anymore. Thanks!

**Comment by LaurentRDC** [comment]
Awesome, I've just released v1.2.1 which contains this fix.

Thank you for reporting this problem, and don't hesitate to open other issues.

**Comment by RyanGibb** [comment]
I'm also experiencing this issue for some pictures. It seems to be that effected images have EXIF orientation metadata, like
```
Orientation                     : Rotate 90 CW
```
(from `exiftool`)

For any future souls the fix was to run:
```
mogrify -auto-orient <filename>
```

From ImageMagick to rotate the underlying image to match the metadata.

**Comment by LaurentRDC** [comment]
@RyanGibb did you experience this issue with the latest version of `hakyll-images`? If so, there may be a problem with the image processing library used by `hakyll-images` (JuicyPixels)

**Comment by RyanGibb** [comment]
Yeah, this was on 1.2.2. I'm happy to share the image if it helps reproduce the issue, or to file a bug upstream.

**Comment by LaurentRDC** [comment]
Please share the image here. I might be doing something dumb.

**Comment by RyanGibb** [comment]
This is the image:

![Image](https://github.com/user-attachments/assets/0608f935-1950-428a-bb0f-4b31efcb6769)

This is the compiler I'm using on it `loadImage >>= scaleImageCompiler 768 1024`

And this is the result:

![Image](https://github.com/user-attachments/assets/e2e8c2bb-da11-4654-9d6f-47a2699771d8)

**Comment by LaurentRDC** [comment]
It's my fault, but the solution is unexpectedly tricky. It might take some time

**Comment by LaurentRDC** [comment]
I think I fixed the problem in 1e68798c4d66af7ae5fad1eeea00a402f05dffa2.

In short, the problem was that EXIF orientation metadata was only preserved when compressing JPEGs, which was the original problem in this thread. EXIF orientation metadata will now be preserved through all operations 

@RyanGibb Would you be able to test the current version of the master branch with your website, before I make a new release?

**Comment by RyanGibb** [comment]
That seems to work! Thanks for the fix :-)

**Comment by LaurentRDC** [comment]
My pleasure. [New release is 1.3.0](https://hackage.haskell.org/package/hakyll-images-1.3.0)
